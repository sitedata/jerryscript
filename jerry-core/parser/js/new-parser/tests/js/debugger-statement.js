/* This test contains debugger statement parser tests. */

next_statement;

debugger;

next_statement;

debugger

next_statement;

{ debugger }

next_statement;
