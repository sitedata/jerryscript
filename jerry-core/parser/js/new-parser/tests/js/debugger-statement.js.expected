
--- Script parsing start ---

  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  0] CBC_EXT_DEBUGGER
  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  0] CBC_EXT_DEBUGGER
  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK
  [  0] CBC_EXT_DEBUGGER
  [  1] CBC_PUSH_LITERAL idx:0->ident(next_statement)
  [  0] CBC_POP_BLOCK

Final byte code dump:

  Maximum stack depth: 1
  Flags: [small_lit_enc]
  Argument range end: 0
  Register range end: 0
  Identifier range end: 1
  Const literal range end: 1
  Literal range end: 1

   0 : CBC_PUSH_LITERAL idx:0(ident)->ident(next_statement)
   2 : CBC_POP_BLOCK
   3 : CBC_EXT_DEBUGGER
   5 : CBC_PUSH_LITERAL idx:0(ident)->ident(next_statement)
   7 : CBC_POP_BLOCK
   8 : CBC_EXT_DEBUGGER
  10 : CBC_PUSH_LITERAL idx:0(ident)->ident(next_statement)
  12 : CBC_POP_BLOCK
  13 : CBC_EXT_DEBUGGER
  15 : CBC_PUSH_LITERAL idx:0(ident)->ident(next_statement)
  17 : CBC_POP_BLOCK
  18 : CBC_RETURN_WITH_BLOCK

Byte code size: 19 bytes

Script parsing successfully completed. Total byte code size: 19 bytes

--- Script parsing end ---

